framework/src/org/checkerframework/common/value/Range.java

```diff
@@ -0,0 +1,324 @@
+package org.checkerframework.common.value;
+
+import java.math.BigInteger;
+
+/**
+ * The Range Class with mathematics operations. Models the range indicated by the @IntRange
```

Comment:
Please document whether Range is allowed to represent the empty set.  If it is not allowed to, it would be nice to have error checking in the constructor.  If it is allowed to, then what is the object invariant or representation invariant of Range?  In particular, what are the from and to values in that case?  It seems that some of the operations are incorrect in that case, such as intersecting two different empty ranges [0..-1] and [100,99].